kore/src/Kore/Strategies/Goal.hs

```diff
@@ -622,11 +626,117 @@ data TransitionRuleTemplate monad goal =
+    = InfoSimplify !goal
+    | InfoRemoveDestination !goal
+    | InfoDeriveSeq ![Rule goal] !goal
+    | InfoDerivePar ![Rule goal] !goal
+
+prettyInfoReachabilityGoal
+    :: Pretty.Pretty goal
+    => Pretty.Doc ann
+    -> goal
+    -> Pretty.Doc ann
+prettyInfoReachabilityGoal transition goal =
+    Pretty.vsep
+        [ "transition:"
+        , transition
+        , "goal:"
+        , Pretty.pretty goal
+        ]
+
+prettyInfoReachabilityGoalAndRules
+    :: Pretty.Pretty goal
+    => Pretty.Pretty rule
+    => Pretty.Doc ann
+    -> goal
+    -> [Rule goal]
+    -> (Rule goal -> rule)
+    -> Pretty.Doc ann
+prettyInfoReachabilityGoalAndRules transition goal rules fromRule =
+    Pretty.vsep
+        [ prettyInfoReachabilityGoal transition goal
+        , "rules:"
+        , Pretty.pretty $ fmap fromRule rules
+        ]
+
+instance Pretty.Pretty (InfoReachability (ReachabilityRule Variable)) where
+    pretty (InfoSimplify goal) =
+        prettyInfoReachabilityGoal "Simplify" goal
+    pretty (InfoRemoveDestination goal) =
+        prettyInfoReachabilityGoal "RemoveDestination" goal
+    pretty (InfoDeriveSeq rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DeriveSeq"
+            goal
+            rules
+            (getRewriteRule . unReachabilityRewriteRule)
+    pretty (InfoDerivePar rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DerivePar"
+            goal
+            rules
+            (getRewriteRule . unReachabilityRewriteRule)
+
+instance Pretty.Pretty (InfoReachability (OnePathRule Variable)) where
+    pretty (InfoSimplify rule) =
+        prettyInfoReachabilityGoal "Simplify" rule
+    pretty (InfoRemoveDestination rule) =
+        prettyInfoReachabilityGoal "RemoveDestination" rule
+    pretty (InfoDeriveSeq rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DeriveSeq"
+            goal
+            rules
+            (getRewriteRule . unRuleOnePath)
+    pretty (InfoDerivePar rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DerivePar"
+            goal
+            rules
+            (getRewriteRule . unRuleOnePath)
+
+instance Pretty.Pretty (InfoReachability (AllPathRule Variable)) where
+    pretty (InfoSimplify rule) =
+        prettyInfoReachabilityGoal "Simplify" rule
+    pretty (InfoRemoveDestination rule) =
+        prettyInfoReachabilityGoal "RemoveDestination" rule
+    pretty (InfoDeriveSeq rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DeriveSeq"
+            goal
+            rules
+            (getRewriteRule . unRuleAllPath)
+    pretty (InfoDerivePar rules goal) =
+        prettyInfoReachabilityGoalAndRules
+            "DerivePar"
+            goal
+            rules
+            (getRewriteRule . unRuleAllPath)
+
+instance Entry (InfoReachability (ReachabilityRule Variable)) where
+    entrySeverity _ = Info
+
+instance Entry (InfoReachability (OnePathRule Variable)) where
```

Comment:
Are the instances for `OnePathRule` and `AllPathRule` actually used? I thought we would only need `ReachabilityRule`.

Reply:
The `Goal` instance of `OnePathRule Variable` uses `transitionRuleTemplate`
which uses `whileSimplify`
which uses `logWhile :: Entry entry => entry -> m a -> m a`